notion-server/src/ethereum/mod.rs

```diff
@@ -403,21 +436,45 @@ impl ShardChain {
     }
 }
 
+trait ToBytes32 {
```

Comment:
I believe another possible approach here would be to create a type `Bytes32` which is just an alias for `[u8; 32]`, then implement a `TryFrom` from `Vec<u8>` (and other types) to convert to `Bytes32`.  @SamWilsn can you chime in on which approach is more idiomatic (or correct me if I'm incorrect here).

Reply:
So... I think creating a newtype would also be a valid option. You'd probably also want to implement `Deref<Target = [u8; 32]>`.

I don't mind doing this as a trait though. Adding a type alias (`type Bytes32 = [u8; 32]`) might be a nice to have as well.

As to which option is more idiomatic... I slightly prefer traits.

Anywhere you accept a `Bytes32` would also be a valid place for `[u8; 32]`, so making a newtype would just complicate APIs.

If we had some special meaning beyond just "an array of 32 bytes", I'd go with a newtype.

Reply:
cool, traits work for me, agree that new types may make things slightly more complex in a non-desirable way.

Reply:
I would be okay with going either way, initially went with `trait` as I was playing around with traits anyway.

Reply:
> cool, traits work for me, agree that new types may make things slightly more complex in a non-desirable way.

Let's stick with this version for now then.